src/targets/gpu/include/migraphx/gpu/context.hpp

```diff
@@ -168,11 +169,40 @@ struct context
         return hip_event_ptr{event};
     }
 
+    value to_value() const
+    {
+        value result;
+        result["events"]  = events.size();
+        result["streams"] = current_device->nstreams();
+
+        return result;
+    }
+
+    void from_value(const value& v)
+    {
+        assert(v.contains("events"));
+        auto v_events = v.at("events");
+        assert(v_events.is_uint64());
```

Comment:
Also this `assert` isn't needed when using `.to` as it will throw an exception if it can't be converted.